### PR TITLE
Add news reader in the launcher

### LINKED ISSUE #536: Allow for reading news in the launcher
### Role

I use PolyMC

### Suggestion

Allow reading news in the launcher, without opening a web page

### Benefit

It would make it less of a chore to read the news

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

When requesting the [news feed xml](https://polymc.org/feed/feed.xml), to display the news to the user, we already receive the news content nicely formatted in HTML.

By opening a new dialog with a text area, and setting its text to the news content, we could display the news nicely formatted in the launcher itself. Note that some HTML parsing may be required, since in one of the news content, there's an `<audio>` tag, so things like that can show up in other ones as well.

This would make it easier to the normal user to read the news, since the process would be much more straight-forward.

### PR DESCRIPTION
**Pull request by flowln**
Closes #536

Clicking the latest news button:
only the latest news post is shown

Clicking the "More news" button:
the latest news post is selected, and there's a list at the left of other news posts

I don't know if this one-click change is worth having two buttons, but removing the "More news" button altogether would also make it harder for the user to realize those buttons are clickable imo, so i think it's fine leaving it as it was before!

### CODE REVIEW

**Review by ZekeZDev — APPROVED** [review]

File: launcher/ui/dialogs/NewsDialog.ui [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+     </item>
+    </layout>
+   </item>
+  </layout>
+ </widget>
+ <resources/>
+ <connections>
+  <connection>
+   <sender>closeButton</sender>
+   <signal>pressed()</signal>
```
Comment (Scrumplex) [inline launcher/ui/dialogs/NewsDialog.ui]:
```suggestion
   <signal>clicked()</signal>
```

`pressed()` triggers on mouse down, which is not quite how buttons generally work :sweat_smile:

**Review by Scrumplex — APPROVED** [review]

File: launcher/ui/dialogs/NewsDialog.ui (hunk 1 above)
Comment (flowln) [reply launcher/ui/dialogs/NewsDialog.ui]: in spite of being an application for making applications, Qt Designer is not the sharpest tool in the shed x.x

Comment (Scrumplex) [reply launcher/ui/dialogs/NewsDialog.ui]: Qt :skull:

**Review by DioEgizio — APPROVED** [review]
seems to work fine to me!

**Review by timoreo22 — APPROVED** [review]

**Pull request merged**